src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/ReferingToObjectAndReassigningItInTheSameStatement.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+{
+    /// <summary>
+    /// ReferenceChanged: Prevent objects from being referenced in statements where they are reassigned
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+
+    public sealed class ReferingToObjectAndReassigningItInTheSameStatement : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "ReferenceChanged";
```

Comment:
TODO: This needs to be fixed.

Reply:
I believe you can use the next available Usage rule ID: `CA2246`. Our last usage rule was [CA2245](https://github.com/MicrosoftDocs/visualstudio-docs/issues/3511)

---
In reply to: [311800142](https://github.com/dotnet/roslyn-analyzers/pull/2717#discussion_r311800142) [](ancestors = 311800142)